applications/SwimmingDEMApplication/custom_utilities/fields/product_of_sines_field.h

```diff
@@ -48,9 +48,7 @@ ProductOfSines(const double period)
     unsigned int number_of_threads = OpenMPUtils::GetNumThreads();
     ResizeVectorsForParallelism(number_of_threads);
 
-    if (period == 0.0){
-        KRATOS_THROW_ERROR(std::invalid_argument, "The period must be non-negative.", 0);
-    }
+    KRATOS_ERROR_IF(period == 0.0) << "The period must be non-negative." << std::endl;
```

Comment:
checking doubles against 0.0 ? ...

Reply:
source [link](https://stackoverflow.com/questions/19837576/comparing-floating-point-number-to-zero)

Maybe kratos has already something similar:
#include <cmath>  /* for std::abs(double) */
inline bool isEqual(double x, double y)
{
  const double epsilon = /* some small number such as 1e-5 */;
  return std::abs(x - y) <= epsilon * std::abs(x);
  // see Knuth section 4.2.2 pages 217-218
}

Coding standards usually recommend against comparing two doubles for exact equality. If you actually want to compare two doubles for exact equality, x == y is the code you want.

Reply:
@farrufat-cimne  I finally used KRATOS_CHECK_DOUBLE_EQUAL(a,b)

Reply:
but It wrong, it's a check with a message. We would need a function which returns true or false... I hace to change it again.